packages/clerk-js/src/core/clerk.ts

```diff
@@ -1003,6 +1013,46 @@ export class Clerk implements ClerkInterface {
     await onAfterSetActive();
   };
 
+  #handlePendingSession = async (session: SignedInSessionResource) => {
+    if (!this.environment) {
+      return;
+    }
+
+    if (session.lastActiveToken) {
+      eventBus.dispatch(events.TokenUpdate, { token: session.lastActiveToken });
+    }
+
+    // Handles multi-session scenario when switching from `active`
+    // to `pending`
+    if (inActiveBrowserTab() || !this.#options.standardBrowser) {
+      await this.#touchCurrentSession(session);
+      session = this.#getSessionFromClient(session.id) ?? session;
+    }
+
+    // Syncs __session and __client_uat, in case the `pending` session
+    // has expired, it needs to trigger a sign-out
+    const token = await session.getToken();
+    if (!token) {
+      eventBus.dispatch(events.TokenUpdate, { token: null });
+    }
+
+    if (session.currentTask) {
+      await navigateToTask(session.currentTask, {
+        globalNavigate: this.navigate,
+        componentNavigationContext: this.#componentNavigationContext,
+        options: this.#options,
+        environment: this.environment,
+      });
+
+      // Reset component navigation context once navigation finishes
+      // to not conflict with after sign-in / after sign-up
+      this.#componentNavigationContext = null;
```

Comment:
I'm surprised to see this value being reset here. What happens if we don't do it?

Reply:
`BaseRouter` only gets unmounted once the application gets closed, or at least if you navigate to page where there's not a single Clerk component 

Since the setter/cleanup logic is happening on `BaseRouter`, there could be a case where we trigger the navigation to tasks once the session updates on the fly (another flow to be handled in a next PR), but `this.#componentNavigationContext` is still defined, therefore it would use it for navigation instead of `Clerk.navigate` that it's important to close opened modals

Reply:
Got it, thanks! I think we need to make sure that `componentNavigationContext` is only defined if a component is defined then, which appears to be the underlying case we're handling here.

Reply:
> if a component is defined then

For `SignIn/SignUp`, makes sense since those are the specific components handling tasks. 

In case we want to leverage this "navigation context" property for other features outside of after sign-in / sign-up in the future, then it makes sense to place it on `BaseRouter`

I don't see a case for the above now, so will try to place it the setter on SignIn/SignUp in a way that the cleanup only happens if the root component gets unmounted